Neolution.Extensions.Identity/TokenSignManager.cs

```diff
@@ -0,0 +1,272 @@
+            var user = await this.userManager.FindByEmailAsync(email);
+            if (user == null)
+            {
+                return null;
+            }
+
+            var signInResponse = await this.signInManager.CheckPasswordSignInAsync(user, password, true);
+            if (signInResponse.Succeeded)
+            {
+                return await this.CreateAccessTokenAsync(user, null);
+            }
+
+            this.logger.LogWarning("Password sign-in for user email={User} failed", email);
+            return null;
+        }
+
+        /// <inheritdoc />
+        public async Task<JsonWebToken?> TwoFactorAuthenticatorSignInAsync(Guid userId, string code, string? authenticationMethod)
+        {
+            this.logger.LogTrace("Perform two factor sign-in for user id={UserId}", userId);
+            var user = await this.userManager.FindByIdAsync(userId);
+            if (user == null)
+            {
+                return null;
+            }
+
+            var error = await this.signInManager.PreSignInCheckAsync(user);
+            if (error != null)
+            {
+                return null;
+            }
+
+            var mfaTokenVerified = await this.userManager.VerifyTwoFactorTokenAsync(user, TokenOptions.DefaultAuthenticatorProvider, code);
+            if (!mfaTokenVerified)
+            {
+                return null;
+            }
+
+            var resetLockoutResult = await this.ResetLockoutWithResultAsync(user);
+            if (!resetLockoutResult.Succeeded)
+            {
+                // If the token is incorrect, record the failure which also may cause the user to be locked out
+                var incrementLockoutResult = await this.userManager.AccessFailedAsync(user) ?? IdentityResult.Success;
+                if (!incrementLockoutResult.Succeeded)
+                {
+                    // Return the same failure we do when resetting the lockout fails after a correct two factor code.
+                    // This is currently redundant, but it's here in case the code gets copied elsewhere.
+                    return null;
+                }
+
+                // ResetLockout got an unsuccessful result that could be caused by concurrency failures indicating an
+                // attacker could be trying to bypass the MaxFailedAccessAttempts limit. Return the same failure we do
+                // when failing to increment the lockout to avoid giving an attacker extra guesses at the two factor code.
+                return null;
+            }
+
+            var claims = new List<Claim>();
+            if (authenticationMethod != null)
+            {
+                claims.Add(new Claim(ClaimTypes.AuthenticationMethod, authenticationMethod));
+            }
+
+            return this.jwtGenerator.GenerateAccessToken(user, claims, "mfa");
+        }
+
+        /// <inheritdoc />
+        public async Task<TUser?> GoogleSignInAsync(string token)
```

Comment:
@neoscie do we still need this? 

I think we should remove this and use the `OpenIdConnectSignInAsync` instead.

Reply:
removed